packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1250,22 +1250,35 @@ abstract class ResidentRunner {
     }
   }
 
-  Future<bool> launchDevTools({bool openInBrowser = false}) async {
+  DevToolsServerAddress activeDevToolsServer() {
+    _devToolsLauncher ??= DevtoolsLauncher.instance;
+    return _devToolsLauncher.activeDevToolsServer;
+  }
+
+  Future<void> serveDevToolsGracefully({
+    Uri devToolsServerAddress,
+    bool openInBrowser = false,
+  }) async {
     if (!supportsServiceProtocol) {
-      return false;
+      return;
+    }
+
+    _devToolsLauncher ??= DevtoolsLauncher.instance;
+    if (devToolsServerAddress != null) {
+      _devToolsLauncher.devToolsUri = devToolsServerAddress;
+    } else {
+      try {
+        return await _devToolsLauncher.serve(openInBrowser: openInBrowser);
+      } catch (e) { // ignore: avoid_catches_without_on_clauses
```

Comment:
Why ignore errors here?

Reply:
We don't want a failed devtools serve to halt a flutter run / attach process, so we swallow the error to fail gracefully. Do you have another suggestion?

Reply:
yes but you are catching all errors, including things like TypeError. What exceptions are you expecting here - could the _devToolsLauncher handle that instead?

Reply:
Friendly ping here - please don't ignore avoid_catches_without_on_clauses

Reply:
removed try catch from resident_runner and pushed exception swallowing into devtools_launcher